### ISSUE TITLE
update and support 3.12

### ISSUE THREAD

**Issue by Zhu-mjl1** [google/latexify_py#206]
## version supported
I've built my project using the latest Python version, 3.12. It would be wonderful if you could update and support 3.12 as well. Thank you in advance!

**Comment by Zhu-mjl1** [comment]
Yes, it occurred right after `pip install latexify-py`, error messages :
```bash
ERROR: Ignored the following versions that require a different python version: 0.0.4 Requires-Python >=3.6, <3.9; 0.0.5 Requires-Python >=3.6, <3.9; 0.0.6 Requires-Python >=3.6, <3.9; 0.0.7 Requires-Python >=3.6, <3.9; 0.1.0 Requires-Python <3.11,>=3.7; 0.1.1 Requires-Python <3.11,>=3.7; 0.2.0 Requires-Python <3.11,>=3.7; 0.2.0b1 Requires-Python <3.11,>=3.7; 0.2.0b2 Requires-Python <3.11,>=3.7; 0.3.0b1 Requires-Python <3.12,>=3.7; 0.3.1 Requires-Python <3.12,>=3.7; 0.4.0 Requires-Python <3.12,>=3.7; 0.4.1 Requires-Python <3.12,>=3.7; 0.4.2 Requires-Python <3.12,>=3.7; 0.4.3.post1 Requires-Python <3.12,>=3.7
ERROR: Could not find a version that satisfies the requirement latexify-py (from versions: none)
ERROR: No matching distribution found for latexify-py
```
Supposed to be version errors ?
By the way, I use macOS so with no idea if it works on the other systems : )

**Comment by EloraCassinyin** [comment]
Same issue...

**Comment by odashi** [comment]
Pip option `--ignore-requires-python` may work, but it would be better to mitigate version requirements in `pyproject.toml`.

**Comment by marvinvanaalst** [comment]
What exactly is holding back lifting the version constraint? According to the [docs](https://docs.python.org/3/whatsnew/3.12.html) the only new syntax feature in 3.12 is the type parameter syntax. Doing a quick check with the version constrained lifted locally and the following snippet works flawlessly:

```python
@latexify.function
def b[T](x: T) -> T:
    return x
```

Happy to help if anything needs to be implemented to get this feature out, as I'd love to be able to use this library in some of my projects that require 3.12. 

Also, since `algorithmic_codegen` already bails on generic `AST` nodes with an `LatexifyNotSupportedError`, will new syntax ever break the library? Or could we just generally support Python >=3.7 and then release updates once missing syntax is required?

**Comment by odashi** [comment]
Sorry for being late to fix this.
There is an ongoing pull request to support newer versions and drop old ones:
https://github.com/google/latexify_py/pull/210

It requires only some tweaks on version-specific behavior, and I think there is basically no serious issues to support newer versions.

**Comment by odashi** [comment]
Released [v0.4.4](https://pypi.org/project/latexify-py/0.4.4/), which involves Python 3.12/3.13 support.
